pkg/apis/intevents/v1alpha1/brokercell_lifecycle.go

```diff
@@ -91,10 +91,21 @@ func (bs *BrokerCellStatus) MarkIngressFailed(reason, format string, args ...int
 // to determine if BrokerCellConditionFanout should be marked as true or
 // false.
 func (bs *BrokerCellStatus) PropagateFanoutAvailability(d *appsv1.Deployment) {
-	if duck.DeploymentIsAvailable(&d.Status, false) {
-		brokerCellCondSet.Manage(bs).MarkTrue(BrokerCellConditionFanout)
-	} else {
-		brokerCellCondSet.Manage(bs).MarkFalse(BrokerCellConditionFanout, "DeploymentUnavailable", "Deployment %q is unavailable.", d.Name)
+	deploymentAvailableFound := false
```

Comment:
nit: Can you make this a helper function and avoid duplicating it many times?
```
func PropagateDeploymentStatus(d *appsv1.Deployment, manager ConditionManager, ct ConditionType)
```

And here you can just call `PropagateDeploymentStatus(d, brokerCellCondSet.manager(bs), BrokerCellConditionFanout `